nasa/cumulus
# CUMULUS-2427: SQS/Kinesis rules should use custom queue

**Summary:** Summary of changes

Addresses [CUMULUS-2427: SQS/Kinesis rules should use custom queue](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2427)

Ran manual tests:

- [x] An SQS rule to confirm that a custom queue (in fact a throttled queue) is used
  - Modify `sqsRuleSpec.js` to add `rule.queueUrl` pointing to the URL of any custom queue, such as the `backgroundProcessing` queue included in each deployment: https://github.com/nasa/cumulus/blob/master/example/spec/parallel/testAPI/sqsRuleSpec.js#L135
  - Confirm that the workflow started by the test has `cumulus_meta.queueUrl` set to the same value from `rule.queueUrl` in the test. This doesn't completely prove that the workflow was started from that queue, so the other alternative is to watch the provided queue for scheduled messages
  - If testing against a throttled queue, verify that the semaphore is incremented to the count of the running workflows (it may be hard to observe this if the workflow runs very quickly and the semaphore is decremented back to 0)
- [x] A Kinesis rule to confirm that a custom queue (in fact a throttled queue) is used
  - Modify `KinesisTestTriggerSpec.js` to add `rule.queueUrl` pointing to the URL of any custom queue, such as the `backgroundProcessing` queue included in each deployment: https://github.com/nasa/cumulus/blob/master/example/spec/parallel/kinesisTests/KinesisTestTriggerSpec.js#L193
  - Confirm that the workflow started by the test has `cumulus_meta.queueUrl` set to the same value from `rule.queueUrl` in the test. This doesn't completely prove that the workflow was started from that queue, so the other alternative is to watch the provided queue for scheduled messages
  - If testing against a throttled queue, verify that the semaphore is incremented to the count of the running workflows (it may be hard to observe this if the workflow runs very quickly and the semaphore is decremented back to 0)

![Screen Shot 2021-03-05 at 3 48 48 PM](https://user-images.githubusercontent.com/1001694/110171918-6a95cc80-7dca-11eb-8ef8-c14fb81c6283.png)

## Changes

* Update `queueMessageForRule` to include other rule properties, such as `queueUrl`, when building workflow messages

## PR Checklist

- [x] Adhoc testing
